File: src/main/java/robTheSpire/actions/StealCardOfTypeAction.java
package robTheSpire.actions;

import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.cards.AbstractCard.CardType;

public class StealCardOfTypeAction extends AbstractCardStealingAction {
    public StealCardOfTypeAction(AbstractCard.CardType type) {
        super(type,(c, prohibited) -> c.color != AbstractCard.CardColor.COLORLESS && c.type != AbstractCard.CardType.CURSE && c.type != AbstractCard.CardType.STATUS);
    }
}

File: src/main/java/robTheSpire/actions/StealCardOfTypeAction.kt
package robTheSpire.actions

import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.cards.AbstractCard.CardType

class StealCardOfTypeAction(type: CardType?) : AbstractCardStealingAction(type, { c: AbstractCard, _: Set<String?>? -> c.color != AbstractCard.CardColor.COLORLESS && c.type != CardType.CURSE && c.type != CardType.STATUS })